semantic_model_generator/sqlgen/generate_sql.py

```diff
@@ -62,6 +73,13 @@ def _return_col_or_expr(
         )
         if expr == "":
             return None
+        # Validate no aggregations in cols.
+        for agg in _AGGREGATION_FUNCTIONS:
+            if agg.lower() in expr.lower():
```

Comment:
just realized this can be quite problematic as user may have column name as max_sales
Let's update to either use appropriate parsing; or check ` avg(` instead of `avg`?

Reply:
we may want to just parse it, as we should also validate if it's a compilable expression